### ISSUE TITLE
Swagger pyang plugin fails for rpc where either "input" or "output" statements are missing

### ISSUE THREAD

**Issue by GermanoSMO** [ict-strauss/COP#4]
This issue is apparently similar to #2 and #3.

According to RFC 6020, the "input" and "output" substatements for an "rpc" yang statement are optional. Therefore, a yang code where an "rpc" does not foresee an "input" (or an "output") statement, is a valid compilable code.

However, with the following code:

``` yang
    grouping dummy {
       leaf alpha {
           type string; }
       leaf beta {
           type string; }
        }

    grouping dummy2 {
       leaf delta {
           type string; }
       leaf epsilon {
           type string; }
        }

    rpc dummyCall {
        description "test";
        input {
            uses dummy;
            }
        }
```

The pyang -f swagger command fails with this error message (similar but not the same as for issue #3, **_a different line number is returned on the last row of the message**_):

> Traceback (most recent call last):
> File "c:/Python34/Scripts/pyang", line 403, in &lt;module&gt; run()
> File "c:/Python34/Scripts/pyang", line 377, in run emit_obj.emit(ctx, modules, fd)
> File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 78, in emit
>        emit_swagger_spec(ctx, modules, fd, ctx.opts.path)
> File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 148, in emit_swagger_spec
>        gen_apis(chs, path, model['paths'], definitions)
> File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 318, in gen_apis
>        gen_api_node(child, path, apis, definitions)
> File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", **_line 369**_, in gen_api_node
>        ref_model[0].arg)}
> IndexError: list index out of range

The same error is returned in the case of an rpc with an "output" substatement but without an "input" substatement. This error should not be returned since constraining an rpc to alwasy support as mandatory both "input" and "output" is not compliant with RFC 6020 Yang's specification.

Python 3.4.3
pyang 1.5
swapper.py as of Nov. 5, 2015 (latest release with #1 fixes)

**Comment by arthurMll** [comment]
Hi GermanoSMO,

I've been working in a patch for the code in order to solve this issues. I generalized the model generation for different yang statements (rpcs, notification, list, container).

I test the code at as far as I saw it fix the issues you mention in #3 and #4.

I encourage to try the code again and see if I miss something. Thanks for you help to fix this bugs and improve the code.

A.

**Comment by GermanoSMO** [comment]
Gracias para corregir el código :) 

These specifics bugs are fixed (so #3 and #4 have been actually addressed successfully and can be closed)-

However, the last code for the swagger plugin now fails on yang codes for which it was previously working. So, apparently this fix has introduced some regressions causing the following final error code from the swagger plugin:
"**UnboundLocalError: local variable  'key' referenced before assignment**"

I'm trying to investigate from my source yang files which could be the yang definition (if any) linked to this error that was never generated before, so that I can open a new issue pin-pointing to the problem.

**Comment by arthurMll** [comment]
Hi Germano,

I noticed the error you mentioned but I think the error is due a "key" missing statement inside a list node which MUST be present following RFC 6020, in fact I noticed that pyang raise an error as well in this case. However, I updated the code in order to raise an exception indicating the problem.

If you found that the problem is reproducer by a correct yang model please post it here and we further investigate it.

A.
